keras_cv/layers/preprocessing/aug_mix.py

```diff
@@ -0,0 +1,310 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import layers
+from keras_cv.utils import preprocessing
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class AugMix(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Performs the AugMix data augmentation technique.  AugMix mixes several
+    chains of augmentations together using weights sampled from a Dirichlet
+    distribution.  A single chain consist of a series of individual augmentations.
```

Comment:
Good docstring, I wonder if we can split it into two sections - a paragraph of high level overview and one with more details

Reply:
Similar to the last comment?

```
    """Performs the AugMix data augmentation technique.

    AugMix....
```

Reply:
Yes but I mean even further, so 3 total paragraphs.

one sentence, one high level overview, one detail heavy description

Reply:
Understood. Can you give me an idea as to what the detail-heavy description should cover? 

Outside the high-level overview, I think the AugMix image (from the paper) is very descriptive. Users that do require that sort of description may be better off just reading the paper (and seeing the image/gif)